src/Open3D/Visualization/Visualizer/GuiVisualizer.cpp

```diff
@@ -1636,6 +1674,20 @@ void GuiVisualizer::SetGeometry(
         impl_->settings_.wgt_point_size->SetEnabled(num_point_clouds > 0);
     }
 
+    if (!impl_->settings_.loaded_materials_.empty()) {
+        int resetIdx = impl_->settings_.wgt_prefab_material->AddItem(
+                MATERIAL_FROM_FILE_NAME.c_str());
+        impl_->settings_.wgt_prefab_material->SetSelectedIndex(resetIdx);
+        ;
+        impl_->settings_.wgt_prefab_material->ChangeItem(
+                (DEFAULT_MATERIAL_NAME + " [default]").c_str(),
+                DEFAULT_MATERIAL_NAME.c_str());
+    } else {
+        impl_->settings_.wgt_prefab_material->ChangeItem(
+                DEFAULT_MATERIAL_NAME.c_str(),
+                (DEFAULT_MATERIAL_NAME + " [default]").c_str());
```

Comment:
move this (identical?) statement out of if and else and do it after for both of them?

Reply:
They aren't identical. One changes from "Default text" -> "Default text [default]" and the other does the reverse.